sensor/kubernetes/eventpipeline/pipeline_impl.go

```diff
@@ -64,6 +70,20 @@ func (p *eventPipeline) ResponsesC() <-chan *message.ExpiringMessage {
 	return p.eventsC
 }
 
+func (p *eventPipeline) stopCurrentContext() {
+	p.contextMtx.Lock()
+	defer p.contextMtx.Unlock()
+	if p.context != nil {
```

Comment:
Not fully sure here, but the `cancelContext` may also be nil - maybe we want to check that for nilness as well?

Reply:
They are always set together. But you're right, this could lead to panic if for some reason they aren't. I changed the `nil` check to be done in `cancelContext` instead.